### ISSUE TITLE
False positive: UnconditionalJumpStatementInLoop with elvis operator ?:

### ISSUE THREAD

**Issue by BraisGabin** [arturbosch/detekt#2388]
## Expected Behavior
The elvis operator is a conditional so code like this should be accepted:

```kotlin
for (resource in Detektor::class.java.classLoader.getResources("META-INF/MANIFEST.MF")) {
    return runCatching { readVersion(resource) }.getOrNull() ?: continue
}
```

## Observed Behavior
It fails

## Context
We found this false positive at #2387

## Your Environment
* Version of detekt used: 1.6.0

**Comment by schalkms** [comment]
Fixed by a9bb93e

**Comment by michalsikora-silvair** [comment]
I'm getting this in `1.18.1`.

```kotlin
fun function(items: List<Int?>): String? {
    for (item in items) {
        val intItem = item ?: continue
        return intItem.toString()
    }
    return null
}
```

**Comment by BraisGabin** [comment]
Yes, please, open a new issue.

**Comment by michalsikora-silvair** [comment]
Will do. :)
